trailblazer/store/base.py

```diff
@@ -61,6 +61,10 @@ def get_filtered_analyses(self, analyses: Query, query: AnalysesRequest) -> Quer
             filters.append(AnalysisFilter.FILTER_BY_STATUSES)
         if query.type:
             filters.append(AnalysisFilter.FILTER_BY_TYPES)
+        if query.case_id:
+            filters.append(AnalysisFilter.FILTER_BY_CASE_ID)
+        if query.most_recent:
+            filters.append(AnalysisFilter.FILTER_BY_MOST_RECENT)
```

Comment:
Could we not rewrite these methods where we have a request DTO in some way like this:

```
filters = []
for field in query.fields():
   if query.field:
       filters.append(SOME ENUM MAGIC FOR CONVERTING)
apply_filters
```

Reply:
I think we can consider some alternative pattern too. I'd like something like a `analyses_request_filters` constant and move the null checks into the filters

Reply:
Leaving this for future PR:s